certify-service/src/main/java/io/mosip/certify/config/AppConfig.java

```diff
@@ -85,11 +88,13 @@ public void run(ApplicationArguments args) throws Exception {
         masterKeyRequest.setReferenceId(org.apache.commons.lang3.StringUtils.EMPTY);
         keymanagerService.generateMasterKey(objectType, masterKeyRequest);
         // TODO: Generate an EC & ED key via K8s Job(INJICERT-469)
-        KeyPairGenerateRequestDto rsaKeyRequest = new KeyPairGenerateRequestDto();
-        rsaKeyRequest.setApplicationId(Constants.CERTIFY_VC_SIGN_RSA);
-        rsaKeyRequest.setReferenceId(Constants.EMPTY_REF_ID);
-        rsaKeyRequest.setForce(false);
-        keymanagerService.generateMasterKey("certificate", rsaKeyRequest);
+        if(pluginMode.equals("DataProvider")) {
+            KeyPairGenerateRequestDto rsaKeyRequest = new KeyPairGenerateRequestDto();
+            rsaKeyRequest.setApplicationId(Constants.CERTIFY_VC_SIGN_RSA);
+            rsaKeyRequest.setReferenceId(Constants.EMPTY_REF_ID);
+            rsaKeyRequest.setForce(false);
+            keymanagerService.generateMasterKey("certificate", rsaKeyRequest);
+        }
```

Comment:
can we not move this code also to below part of same if condition ?
I believe this key and ed key are created freshly in this release.

Reply:
Moved RSA key generation to the condition below